modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java

```diff
@@ -610,14 +618,53 @@ public boolean attemptReconnection()
         return false;
     }
 
-    public ProtocolVersion makeBrokerConnection(BrokerDetails brokerDetail) throws IOException, AMQException
+    public ProtocolVersion makeBrokerConnection(final BrokerDetails brokerDetail) throws IOException, AMQException
     {
-        return _delegate.makeBrokerConnection(brokerDetail);
+        // Requires permission java.lang.RuntimePermission "modifyThreadGroup"
+        Object returnObject = AccessController.doPrivileged(new PrivilegedAction() {
+            public Object run() {
+                try {
+                    return _delegate.makeBrokerConnection(brokerDetail);
+                } catch (IOException | AMQException e) {
+                    return e;
+                }
+            }
+        });
+
+        if (returnObject instanceof ProtocolVersion) {
+            return (ProtocolVersion) returnObject;
+        } else if (returnObject instanceof IOException) {
+            throw (IOException) returnObject;
+        } else if (returnObject instanceof AMQConnection) {
+            throw (AMQException) returnObject;
+        } else {
+            return null;
+        }
+
     }
 
-    public <T, E extends Exception> T executeRetrySupport(FailoverProtectedOperation<T,E> operation) throws E
+    public <T, E extends Exception> T executeRetrySupport(final FailoverProtectedOperation<T,E> operation) throws E
     {
-        return _delegate.executeRetrySupport(operation);
+        // Requires permission java.lang.RuntimePermission "modifyThreadGroup"
+        Object returnObject = AccessController.doPrivileged(new PrivilegedAction() {
```

Comment:
I think we can use PrivilegedExceptionAction[1] for that purpose.
[1]. http://docs.oracle.com/javase/7/docs/api/java/security/PrivilegedExceptionAction.html